lib/_http_client.js

```diff
@@ -133,10 +133,12 @@ function ClientRequest(options, cb) {
   if (self.socketPath) {
     self._last = true;
     self.shouldKeepAlive = false;
-    const optionsPath = {
-      path: self.socketPath
+    const createConnectionOptions = {
+      path: self.socketPath,
+      socketTimeout: options.socketTimeout
```

Comment:
I think renaming to `timeout` would work. The main reason why I thought about naming it `socketTimeout` (or even `connectionTimeout`) is that we are also able to set the timeout via the `setTimeout` function after the `socket` \ `request` is created and as a user it wouldn't be immediately obvious why there are those two ways to provide a timeout. Hence, I tried to make it a little more clear with naming it `socketTimeout` instead of just `timeout`. Tbh, though, I am not sure if it really makes much of a difference.